component/main.jsonnet

```
@@ -34,14 +34,27 @@ local ns_patch =
     }
   );
 
+local transformRelabelConfigs(remoteWriteConfig) = if std.member(std.objectFields(remoteWriteConfig), 'writeRelabelConfigs')
+then remoteWriteConfig {
+  writeRelabelConfigs: std.map(
+    function(wrlc) { [key]: wrlc[key] for key in std.objectFields(wrlc) if key != 'timeseries' } {
+      [if std.member(std.objectFields(wrlc), 'timeseries') && std.length(wrlc.timeseries) > 0
+      then 'regex'
+      else null]: std.format('(%s)', std.join('|', wrlc.timeseries)),
+    },
```

This could be written as follows (untested):

```suggestion
    function(wrlc) wrlc {
      timeseries:: [], // make field timeseries hidden so it doesn't get emitted in the Jsonnet output
      [if std.objectHas(wrlc, 'timeseries') && std.length(wrlc.timeseries) > 0 then 'regex']: 
        std.format('(%s)', std.join('|', wrlc.timeseries)),
    },
```